yarn-project/foundation/src/log/pino-logger.ts

```diff
@@ -11,14 +11,23 @@ import { getLogLevelFromFilters, parseEnv } from './log-filters.js';
 import { type LogLevel } from './log-levels.js';
 import { type LogData, type LogFn } from './log_fn.js';
 
-export function createLogger(module: string): Logger {
+export function createLogger(module: string, fixedTerms = {}): Logger {
   module = logNameHandlers.reduce((moduleName, handler) => handler(moduleName), module.replace(/^aztec:/, ''));
   const pinoLogger = logger.child({ module }, { level: getLogLevelFromFilters(logFilters, module) });
 
+  // Only perform copy of data if fixed terms are provided
```

Comment:
Used in the libp2p logger to enable tagging the sourcePeer to all messages. If fixed terms are not set, then we should skip doing the extra allocation in the ternary operation below